### ISSUE TITLE
InconsistentAnalysisException after upgrading to 2.0.3

### ISSUE THREAD

**Issue by noinskit** [rrousselGit/freezed#671]
**Describe the bug**
After I upgraded freezed from 1.1.1 to 2.0.3+1 (and at the same time transitively analyzer from 3.4.1 to 4.1.0), I started getting non-deterministic `InconsistentAnalysisException` exceptions on `flutter pub run build_runner build`.

It does not happen every time and not always with the same set of files, also tends to differ e.g. between my laptop and my CI/CD env. Example error logs:
```
[SEVERE] freezed:freezed on lib/src/population/diet_score_population_features.dart:
InconsistentAnalysisException: Requested result might be inconsistent with previously returned results
```

I tried to dive into analyzer-4.1.0 source code and the `_checkConsistency()` check throws due to nonempty "pendingFileChanges", most of them not using freezed or other code generation.

**To Reproduce**
I was not able to create a minimal reproducible example, it's happening for me in a real-life project and preventing me from upgrading freezed.

**Expected behavior**
Consistent successes, not failures.

**Comment by rrousselGit** [comment]
The fact that there's no Freezed mention in the stacktrace could mean that this is a source_gen/build_resolvers/analyzer issue

**Comment by rrousselGit** [comment]
I'd look into source_gen's _Builder.build

I don't think Freezed is even executed at all here. This stacktrace seems to be about the steps before Freezed executes

**Comment by noinskit** [comment]
FWIW looks like I'm able to work this around for now by downgrading analyzer to 3.4.1 in dev_dependencies.

**Comment by noinskit** [comment]
Sorry for the false alarm, this was a bug in build_resolvers (triggered by analyzer-4.0.0) and it's already fixed in build_resolvers 2.0.9.
